book/docs/setup/MacOS.md

```diff
@@ -138,4 +138,63 @@ Open your finest browser and navigate to:
 
 Cloned from [Reddit](https://www.reddit.com/r/cardano/comments/mmzut6/macos_plutus_playground_build_instructions/)
 
-Go give u/RikAlexander karma!
\ No newline at end of file
+Go give u/RikAlexander karma!
+
+
+
+## Troubleshooting
+
+
+### `nix-shell` exits with segmentation fault
+
+Error message:
+```bash
+$ nix-shell
+[1]    296 segmentation fault  nix-shell
+```
+
+Solutions:
+- Comment out the line `withHoogle = false;` in `shell.nix` before running `nix-shell` like
+```nix
+#withHoogle = false;
+```
+- Better: Switch to master branch, this issue (along with others) should be solved.
+  E.g. the following fixes some more issues with macOS Big Sur
+    - Commit: 34aa9c323ed6da68a11f41d41d5aca9f469aaf4b
+    - Date: 23.04.2021
+
+
+### `haskell-language-server` fails with segmentation fault / You do not get any Haskell editor-integration working
+
+- Problem: The problem described here boils down to running `haskell-language-server` in one of the plutus-pioneer-program repos and `haskell-language-server` fails with a segmentation fault
+```bash
+$ cd plutus-pioneer-program/code/week01
+$ haskell-language-server
+haskell-language-server version: 0.9.0.0 ...
+...
+[INFO] Using interface files cache dir: ghcide
+[INFO] Making new HscEnv[plutus-pioneer-program-week01-0.1.0.0-inplace]
+Segmentation fault: 11
+```
+
+Solution:
+- Upgrade the `plutus` repo to a later release (maybe master branch).
+  Since `haskell-language-server` is in version 0.9.0.0 (as you can see in the first line after execution) and this version is not ready for macOS Big Sur.
```

Comment:
```suggestion
  Since `haskell-language-server` has version 0.9.0.0 (as you can see in the first line after execution) and this version is not ready for macOS Big Sur.
```